src/test/optimizer/strategy/between_composition_rule_test.cpp

```diff
@@ -118,7 +118,7 @@ TEST_F(BetweenCompositionTest, EmptyColumnRange) {
   EXPECT_LQP_EQ(result_lqp, expected_lqp);
 }
 
-TEST_F(BetweenCompositionTest, OrExpression) {
+TEST_F(BetweenCompositionTest, NoPullPastOrExpression) {
```

Comment:
I don't see a change to the BetweenCompositionRule, so how was #1813 fixed? Isn't it still broken on certain diamonds?

Reply:
Good that you mention it. I almost assumed that the problem was fixed because nothing crashed in the benchmarks again.
The actual problem was not that the `BetweenCompositionRule` could not handle diamond LQPs but that it could not handle expressions like `a = 1 AND b = 2`. The crash happened because the `DisjunctionToUnionRule` created two of those expressions for TPC-H 7.

Reply:
Exactly.